### PR TITLE
optimize(hz): optimize the use experience for hz client

### PR DESCRIPTION
**Pull request by FGYFFFF**
#### What type of PR is this?

optimize
#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) More detail description for this PR(en: English/zh: Chinese).

en: optimize the use experience for hz client
zh(optional): 根据最近的反馈优化 hz client 的使用体验

#### Which issue(s) this PR fixes:

- [x] int value to string automatically
- [x] configuration of the default client
- [x] more concise client generation path (used as SDK)
- [x] "GET" requests add parameters to the query by default (when there is no tag)
- [x] use "GetName" for thrift

### CODE REVIEW

**Comment by FGYFFFF** [comment]
@welkeyever @li-jin-gou @Duslia @wzekin  cc

File: cmd/hz/generator/package_tpl.go [hunk 1]
```diff
@@ -588,9 +588,9 @@ func (r *request) setError(err interface{}) *request {
 	return r
 }
 
-func (r *request) setHeaders(headers map[string]string) *request {
+func (r *request) setHeaders(headers map[string]interface{}) *request {
```
Comment (welkeyever) [inline cmd/hz/generator/package_tpl.go:591]: 这个map从哪里生成的，现在还有string/int共存的情况？

File: cmd/hz/app/app.go [hunk 2]
```diff
@@ -163,6 +163,7 @@ func Init() *cli.App {
 	routerDirFlag := cli.StringFlag{Name: "router_dir", Usage: "Specify the router relative path (based on \"out_dir\").", Destination: &globalArgs.RouterDir}
 	baseDomainFlag := cli.StringFlag{Name: "base_domain", Usage: "Specify the request domain.", Destination: &globalArgs.BaseDomain}
 	clientDirFlag := cli.StringFlag{Name: "client_dir", Usage: "Specify the client path. If not specified, IDL generated path is used for 'client' command; no client code is generated for 'new' command", Destination: &globalArgs.ClientDir}
+	forceClientDirFlag := cli.StringFlag{Name: "force_client_dir", Usage: "Specify the client path, and do not use namespace as a subpath", Destination: &globalArgs.ForceClientDir}
```
Comment (welkeyever) [inline cmd/hz/app/app.go:166]: `and won't use namespaces as subpaths`

File: cmd/hz/generator/package_tpl.go (hunk 1 above)
Comment (FGYFFFF) [reply cmd/hz/generator/package_tpl.go:591]:
```
type TestStruct struct {
         HeaderTag int
         QueryTag int
         PathTag int
}
```
例如，以上的类型的参数均有可能被设置为"数值型"的类型。因此需要转为 "string"，要不然无法编译

File: cmd/hz/app/app.go (hunk 2 above)
Comment (FGYFFFF) [reply cmd/hz/app/app.go]: done

**Review by Duslia — DISMISSED** [review]

File: cmd/hz/generator/package_tpl.go (hunk 1 above)
Comment (welkeyever) [reply cmd/hz/generator/package_tpl.go:591]: 有点奇怪，这个map[string]string 和机构体是咋映射的？为啥不是传入的时候直接将结构体转为map[string]string呢？

Comment (FGYFFFF) [reply cmd/hz/generator/package_tpl.go:591]: 读结构体的 tag，然后根据不同的 tag，将结构体的 field 分发到 query、header、form 等的设置函数，

Comment (FGYFFFF) [reply cmd/hz/generator/package_tpl.go]: done

**Review by Duslia — DISMISSED** [review]

**Review by welkeyever — APPROVED** [review]

**Review by Duslia — APPROVED** [review]

**Pull request merged**